test/integration/resourceclaim/feature_enable_disable_test.go

```diff
@@ -0,0 +1,221 @@
+		},
+		etcdOptions)
+	client2, err := clientset.NewForConfig(server2.ClientConfig)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	rcEnabledName := "test-enable-dra-resourceclaim-device-status-rc-enabled"
+	rcEnabled := &v1alpha3.ResourceClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: rcEnabledName,
+		},
+		Spec: v1alpha3.ResourceClaimSpec{
+			Devices: v1alpha3.DeviceClaim{
+				Requests: []v1alpha3.DeviceRequest{
+					{
+						Name:            "bar",
+						DeviceClassName: "bar",
+						Count:           1,
+						AllocationMode:  v1alpha3.DeviceAllocationModeExactCount,
+					},
+				},
+			},
+		},
+	}
+
+	if _, err := client2.ResourceV1alpha3().ResourceClaims(ns.Name).Create(context.TODO(), rcEnabled, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	// Tests the validation is enabled.
+	// validation will refuse this update as the device is not allocated.
+	rcEnabled.Status = v1alpha3.ResourceClaimStatus{
+		Devices: []v1alpha3.AllocatedDeviceStatus{
+			{
+				Driver: "bar",
+				Pool:   "bar",
+				Device: "bar",
+				Data: runtime.RawExtension{
+					Raw: []byte(`{"kind": "foo", "apiVersion": "dra.example.com/v1"}`),
+				},
+				NetworkData: &v1alpha3.NetworkDeviceData{
+					InterfaceName: "net-1",
+					Addresses: []string{
+						"10.9.8.0/24",
+						"2001:db8::/64",
+					},
+					HWAddress: "ea:9f:cb:40:b1:7b",
+				},
+			},
+		},
+	}
+	if _, err := client2.ResourceV1alpha3().ResourceClaims(ns.Name).UpdateStatus(context.TODO(), rcEnabled, metav1.UpdateOptions{}); err == nil {
+		t.Fatalf("Expected error (must be an allocated device in the claim)")
+	}
+
+	rcEnabled.Status = v1alpha3.ResourceClaimStatus{
+		Allocation: &v1alpha3.AllocationResult{
+			Devices: v1alpha3.DeviceAllocationResult{
+				Results: []v1alpha3.DeviceRequestAllocationResult{
+					{
+						Request: "bar",
+						Driver:  "bar",
+						Pool:    "bar",
+						Device:  "bar",
+					},
+				},
+			},
+		},
+		Devices: []v1alpha3.AllocatedDeviceStatus{
+			{
+				Driver: "bar",
+				Pool:   "bar",
+				Device: "bar",
+				Data: runtime.RawExtension{
+					Raw: []byte(`{"kind": "foo", "apiVersion": "dra.example.com/v1"}`),
+				},
+				NetworkData: &v1alpha3.NetworkDeviceData{
+					InterfaceName: "net-1",
+					Addresses: []string{
+						"10.9.8.0/24",
+						"2001:db8::/64",
+					},
+					HWAddress: "ea:9f:cb:40:b1:7b",
+				},
+			},
+		},
+	}
+	if _, err := client2.ResourceV1alpha3().ResourceClaims(ns.Name).UpdateStatus(context.TODO(), rcEnabled, metav1.UpdateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	// Tests the field is enabled.
+	rcEnabled, err = client2.ResourceV1alpha3().ResourceClaims(ns.Name).Get(context.TODO(), rcEnabledName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(rcEnabled.Status.Devices) != 1 {
+		t.Fatalf("expected 1 Device in status got %d", len(rcEnabled.Status.Devices))
+	}
+
+	// shutdown apiserver with the feature enabled
+	server2.TearDownFn()
+}
```

Comment:
@liggitt @thockin let me know if it makes sense to add this additional step that validates that reading from an apiserver with the feature disabled keeps the gated fields

```diff
diff --git a/test/integration/resourceclaim/feature_enable_disable_test.go b/test/integration/resourceclaim/feature_enable_disable_test.go
index 526d04fc119..6fb46ec8664 100644
--- a/test/integration/resourceclaim/feature_enable_disable_test.go
+++ b/test/integration/resourceclaim/feature_enable_disable_test.go
@@ -49,6 +49,8 @@ func TestEnableDisableDRAResourceClaimDeviceStatus(t *testing.T) {
        if err != nil {
                t.Fatalf("Unexpected error: %v", err)
        }
+       // shutdown apiserver with the feature disabled
+       defer server1.TearDownFn()
 
        ns := framework.CreateNamespaceOrDie(client1, "test-enable-dra-resourceclaim-device-status", t)
 
@@ -108,9 +110,6 @@ func TestEnableDisableDRAResourceClaimDeviceStatus(t *testing.T) {
                t.Fatalf("expected 0 Device in status got %d", len(rcDisabled.Status.Devices))
        }
 
-       // shutdown apiserver with the feature disabled
-       server1.TearDownFn()
-
        // apiserver with the feature enabled
        server2 := kubeapiservertesting.StartTestServerOrDie(t, apiServerOptions,
                []string{
@@ -218,4 +217,13 @@ func TestEnableDisableDRAResourceClaimDeviceStatus(t *testing.T) {
 
        // shutdown apiserver with the feature enabled
        server2.TearDownFn()
+
+       // Tests the field is still present when reading from an apiserver with the feature disabled
+       rcEnabled, err = client1.ResourceV1alpha3().ResourceClaims(ns.Name).Get(context.TODO(), rcEnabledName, metav1.GetOptions{})
+       if err != nil {
+               t.Fatal(err)
+       }
+       if len(rcEnabled.Status.Devices) != 1 {
+               t.Fatalf("expected 1 Device in status got %d", len(rcEnabled.Status.Devices))
+       }
 }
```